src/lib/common/include/sol-mainloop.h

```diff
@@ -161,6 +161,7 @@ struct sol_main_callbacks {
         return sol_mainloop_default_main(&(CALLBACKS), argc, argv);  \
     }
 
+#ifndef SOL_PLATFORM_CONTIKI
```

Comment:
this is not right, we should have SOL_MAIN_DEFAULT() always defined and working for every platform. I guess for contiki it would have a different implementation, but we should define it.